packages/compiler-cli/src/ngtsc/annotations/src/component.ts

```diff
@@ -304,6 +318,38 @@ export class ComponentDecoratorHandler implements
     return output;
   }
 
+  index(context: IndexingContext, node: ClassDeclaration, analysis: ComponentHandlerData) {
+    // The component template may have been previously parsed without preserving whitespace or with
+    // `leadingTriviaChar`s, both of which may manipulate the AST into a form not representative of
+    // the source code, making it unsuitable for indexing. The template is reparsed with preserving
+    // options to remedy this.
+    const template = analysis.parseTemplate({
+      preserveWhitespaces: true,
+      leadingTriviaChars: [],
+    });
+    const scope = this.scopeRegistry.getScopeForComponent(node);
+    const selector = analysis.meta.selector;
+    const matcher = new SelectorMatcher<DirectiveMeta>();
+    if (scope !== null) {
+      for (const directive of scope.compilation.directives) {
+        const {selector} = directive;
+        matcher.addSelectables(CssSelector.parse(selector), directive);
```

Comment:
```suggestion
        matcher.addSelectables(CssSelector.parse(directive.selector), directive);
```

Then you don't need the extraction of `selector` to a variable.